apps/web/src/components/WalletConnect/SessionProposalModal.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  Box,
+  Card,
+  FormControl,
+  FormErrorMessage,
+  HStack,
+  ModalContent,
+  Text,
+  VStack,
+  useToast,
+} from "@chakra-ui/react";
+import { type WalletKitTypes } from "@reown/walletkit";
+import { useDynamicModalContext } from "@umami/components";
+import { useAsyncActionHandler, useGetImplicitAccount, walletKit } from "@umami/state";
+import { buildApprovedNamespaces, getSdkError } from "@walletconnect/utils";
+import { useState } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+
+import RequestModal from "./RequestModal";
+import { CheckmarkIcon, CloseIcon } from "../../assets/icons";
+import { OwnedImplicitAccountsAutocomplete } from "../AddressAutocomplete";
+
+/**
+ * Finds the overlap between the chains supported by the wallet and the chains requested by the dApp.
+ * We take the first required chain as Umami supports connection for only one chain per request.
+ */
+function getChainToConfirm(required: (string | string[] | undefined)[]) {
+  // take the first required chain which is supported by the wallet
+  for (const chain of required.flat()) {
+    if (chain && ["tezos:mainnet", "tezos:ghostnet"].includes(chain)) {
+      return chain;
+    }
+  }
+  return undefined;
+}
+
+export default function SessionProposalModal({
+  proposal,
+}: {
+  proposal: WalletKitTypes.SessionProposal;
+}) {
+  console.log("SessionProposalModal", proposal);
+  const getAccount = useGetImplicitAccount();
+  const toast = useToast();
+
+  const { onClose } = useDynamicModalContext();
+  const { handleAsyncAction } = useAsyncActionHandler();
+
+  const [isLoadingApprove, setIsLoadingApprove] = useState(false);
+  const [isLoadingReject, setIsLoadingReject] = useState(false);
+  const form = useForm<{ address: string }>({
+    mode: "onBlur",
+  });
+  const {
+    getValues,
+    formState: { errors, isValid },
+  } = form;
+
+  const required = Object.entries(proposal.params.requiredNamespaces).map(([key, values]) =>
+    key.includes(":") ? key : values.chains
+  );
+
+  if (required.length !== 1) {
+    console.error("Expected only one required chain", required);
+  }
+
+  const chain = getChainToConfirm(required);
+  if (!chain) {
+    console.error("No supported chain found", required);
+  }
+
+  const onApprove = () =>
+    handleAsyncAction(async () => {
+      const account = getAccount(getValues().address);
+
+      try {
+        if (!chain) {
```

Comment:
Why the same chain check here again? Would it be possible to only have it in one place?

Reply:
previously I had to check if chain is known to check if it's available. Now I need to decide what to show in the modal

Reply:
Ah, it's just console.error(), not throwing an error, got it now!

@dianasavvatina 
Not for this PR probably, but it would be good to check what is the expected behaviour on chains mismatch case (when the request would be rejected on the WalletConnect side) - do we want to show SessionProposalModal, just an error toast or nothing at all?

Reply:
I've seen the error in toast